c++/src/kj/compat/http.c++

```diff
@@ -5210,7 +5210,8 @@ public:
     });
   }
 
-  ConnectRequest connect(kj::StringPtr host, const HttpHeaders& headers) override {
+  ConnectRequest connect(
+      kj::StringPtr host, const HttpHeaders& headers, ConnectSettings settings) override {
```

Comment:
It appears this version of `connect()` (for `HttpClientImpl`, which makes a `CONNECT` request) ignores the `useTls` setting, which I think people would find surprising.

I think the right answer here is to allow an optional `TlsContext` to be provided when creating the client, and automatically use it to upgrade the connection once established. This would match regular HTTP proxy conventions. If the caller doesn't actually want TLS to apply at this level, and instead wants the destination proxy server to deal with it, it would be up to the caller to rewrite the setting into a header or whatever.

You could also KJ_UNIMPLEMENTED for now.